cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -32,26 +35,31 @@
  */
 public class BasicWorkflowToken implements WorkflowToken {
   private Map<String, Map<String, Long>> mapReduceCounters;
-  private final Map<String, List<NodeValueEntry>> tokenValueMap = Maps.newHashMap();
+  private final Table<Scope, String, List<NodeValueEntry>> tokenValueMap = HashBasedTable.create();
```

Comment:
nit: Since this is not a ``Map`` anymore, maybe call it ``tokenValueTable``? Or even just ``data``?